farmhelper-mod/src/main/java/com/jelly/farmhelper/features/PetSwapper.java

```diff
@@ -0,0 +1,223 @@
+package com.jelly.farmhelper.features;
+
+import com.jelly.farmhelper.config.Config;
+import com.jelly.farmhelper.macros.MacroHandler;
+import com.jelly.farmhelper.utils.Clock;
+import com.jelly.farmhelper.utils.LogUtils;
+import com.jelly.farmhelper.utils.PlayerUtils;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.inventory.GuiChest;
+import net.minecraft.inventory.ContainerChest;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.StringUtils;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.common.gameevent.TickEvent;
+
+public class PetSwapper {
+
+    private static Minecraft mc = Minecraft.getMinecraft();
+    public String swappedPet;
+    private String originalPet;
+    private static Clock delay = new Clock();
+
+    //credit to Nirox for the clicktypes
+    public enum ClickType {
+        PICKUP,
+        QUICK_MOVE,
+        SWAP,
+        CLONE,
+        THROW,
+        QUICK_CRAFT
+    }
+
+    public enum StatePet {
+        OPEN_MENU,
+        FIND_PET,
+        TURN_PAGE,
+        CLICK_PET,
+        CLOSE_MENU
+
+    }
+
+    private static StatePet state = StatePet.OPEN_MENU;
+    private static ContainerChest chest;
+    private static int petSlot;
+    private int pageSlot;
+    private boolean shouldSwap = false;
+
+    public void swapPets() {
```

Comment:
Maybe you could make this a static method instead of having to create an object to call it